src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -614,14 +615,24 @@ private static void parseSingleAltAllele(List<Allele> alleles, String alt, int l
     }
 
     public final static boolean canDecodeFile(final String potentialInput, final String MAGIC_HEADER_LINE) {
+    	FileInputStream fis = null;
+    	GZIPInputStream giz = null;
+    	BlockCompressedInputStream bcis = null;
         try {
-            return isVCFStream(new FileInputStream(potentialInput), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new GZIPInputStream(new FileInputStream(potentialInput)), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new BlockCompressedInputStream(new FileInputStream(potentialInput)), MAGIC_HEADER_LINE);
+        	fis = new FileInputStream(potentialInput);
```

Comment:
I've previous raised #494 to address this in a more general context as there are a number of places that I don't touch that have the same problems and to do it properly requires a code review of all opened streams. Should I be performing these sort of bug fixes which are not directly related to the performance enhancements in this PR?